src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -928,6 +911,31 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
             noway_assert(structSize <= MAX_PASS_MULTIREG_BYTES);
 #endif // TARGET_ARM64
 
+            unsigned roundedDstSize = treeNode->GetStackByteSize();
+            if (roundedDstSize != structSize)
+            {
+                unsigned roundedSrcSize = structSize;
+                // We can generate a smaller code if store size is a multiple of TARGET_POINTER_SIZE.
+                // The dst size can be rounded up to PUTARG_STK size.
+                // The src size can be rounded up if it reads a local variable slot because the local
+                // variable stack allocation size is rounded up to be a multiple of the slot size.
```

Comment:
```suggestion
                // variable stack allocation size is rounded up to be a multiple of the TARGET_POINTER_SIZE.
```
will fix before the merge.